docs/platform-engineers/components/custom-component.md

```diff
@@ -8,95 +8,164 @@ In this section, it will introduce how to use [CUE](https://cuelang.org/) to dec
 
 ## Declare `ComponentDefinition`
 
-Here is a CUE based `ComponentDefinition` example which provides a abstraction for stateless workload type:
+First, generate `ComponentDefinition` scaffolds via `vela def init`:
 
-```yaml
-apiVersion: core.oam.dev/v1beta1
-kind: ComponentDefinition
-metadata:
-  name: stateless
-spec:
-  workload:
-    definition:
-      apiVersion: apps/v1
-      kind: Deployment
-  schematic:
-    cue:
-      template: |
-        parameter: {
-        	name:  string
-        	image: string
-        }
-        output: {
-        	apiVersion: "apps/v1"
-        	kind:       "Deployment"
-        	spec: {
-        		selector: matchLabels: {
-        			"app.oam.dev/component": parameter.name
-        		}
-        		template: {
-        			metadata: labels: {
-        				"app.oam.dev/component": parameter.name
-        			}
-        			spec: {
-        				containers: [{
-        					name:  parameter.name
-        					image: parameter.image
-        				}]
-        			}
-        		}
-        	}
-        }
+```shell
+vela def init stateless -t component -o stateless.cue
 ```
+
+It generates a file:
+
+```shell
+$ cat stateless.cue
+stateless: {
+	annotations: {}
+	attributes: workload: definition: {
+		apiVersion: "<change me> apps/v1"
+		kind:       "<change me> Deployment"
+	}
+	description: ""
+	labels: {}
+	type: "component"
+}
+
+template: {
+	output: {}
+	parameter: {}
+}
+```
+
 In detail:
 - `.spec.workload` is required to indicate the workload type of this component.
 - `.spec.schematic.cue.template` is a CUE template, specifically:
     * The `output` filed defines the template for the abstraction.
     * The `parameter` filed defines the template parameters, i.e. the configurable properties exposed in the `Application`abstraction (and JSON schema will be automatically generated based on them).
 
-Let's declare another component named `task`, i.e. an abstraction for run-to-completion workload.
+Edit this auto-generated custom component file:
```

Comment:
Please declare how this auto generated

Reply:
It is declared above: https://github.com/oam-dev/kubevela.io/pull/227/files/4e51ebdc1e931bba1af116455f361bdd272ba0f3#diff-47a91ec8d3a3053891899d042dc8553a22685e02d4f931f5737c98fc0aab66a3R11

Should we describe it more specifically?

Reply:
yes, you should generate it with command that can be copy-paste and work